### DISCUSSION TITLE
Incremental builds and custom source

### DISCUSSION THREAD

**Discussion by vladimir-stornest** [gatsbyjs/gatsby#29806] [Help] [answered]
Hello,

I have an application that sources data from custom GraphQL CMS written in node.js. I want to start using Incremental Builds. 

I'm trying to find official documentation on implementing my sourcing to support Incremental builds for my custom API. I checked [this article](https://support.gatsbyjs.com/hc/en-us/articles/360052503494-Developing-a-first-class-CMS-integration-for-Gatsby-Cloud#step-1-source-plugin-required), but it references these links which don't include the needed implementation for Incremental Builds either:
https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/creating-a-source-plugin/#create-foreign-key-relationships-between-data
https://www.gatsbyjs.com/docs/creating-a-source-plugin/

**Comment by KyleAMathews** [comment, marked as answer]
Incremental builds work automatically.

The main thing you should do to speed up builds is to support delta fetched. So on subsequent runs, only fetch updated data https://www.gatsbyjs.com/docs/creating-a-source-plugin/#improve-plugin-developer-experience-by-enabling-faster-sync
